File: backend/translate.py
from flask import Flask, request, jsonify
from googletrans import Translator, LANGUAGES

app = Flask(__name__)
translator = Translator()

@app.route('/translate', methods=['POST'])
def translate():
    data = request.json
    language = data.get('language')
    text = data.get('text')

    if not language or not text:
        return jsonify({'error': 'Missing language or text in request body'}), 400

    if language not in LANGUAGES:
        return jsonify({'error': 'Invalid language code'}), 400

    translated_text = translator.translate(text, dest=language).text
    return jsonify({'translated_text': translated_text})

if __name__ == '__main__':
    app.run(debug=True)

File: backend/routes/translate.js
const express = require("express");
const router = express.Router();
const axios = require("axios");

router.post('/translate', async (req, res) => {
    try {
        const { language, text } = req.body;

        // Make a POST request to the Flask translation server
        const response = await axios.post('http://localhost:5000/translate', { language, text });

        // Send back the translated text received from Flask server
        res.json({ translated_text: response.data.translated_text });
    } catch (error) {
        console.error('Error:', error);
        res.status(500).json({ error: 'Internal server error' });
    }
});

module.exports = router;